app/models/host.rb

```diff
@@ -91,6 +91,7 @@ class Host < ApplicationRecord
   scope :with_failed_rules_count, lambda {
     sq = Host.left_outer_joins(test_result_profiles: :rule_results)
              .where(rule_results: { result: RuleResult::FAILED })
+             .or(Host.where(rule_results: { id: nil }))
              .select(arel_table[:id].as('id'), RuleResult.arel_table[:result].count.as('rules_failed'))
              .group('hosts.id')
```

Comment:
Does the SQL around the `COUNT("rule_results"."result")` part look correct to you? @skateman 
```
SELECT "inventory"."hosts"."id" AS id, COUNT("rule_results"."result") AS rules_failed FROM "inventory"."hosts"
LEFT OUTER JOIN "test_results" ON "test_results"."host_id" = "inventory"."hosts"."id" LEFT OUTER JOIN 
"profiles" ON "profiles"."id" = "test_results"."profile_id" LEFT OUTER JOIN "test_results" 
"test_results_profiles_join" ON "test_results_profiles_join"."profile_id" = "profiles"."id" LEFT OUTER JOIN 
"rule_results" ON "rule_results"."test_result_id" = "test_results_profiles_join"."id" WHERE ("rule_results"."result" 
IN ($1, $2, $3, $4) OR "rule_results"."id" IS NULL) GROUP BY "hosts"."id"  [["result", "fail"], ["result", "error"], 
["result", "unknown"], ["result", "fixed"]]
```

I tested some more and realized that the rules_failed count is too high sometimes. At first I thought my test data wasn't right but I don't think that's the issue. 🤔

Reply:
You are doing sorting on the host against ALL its profiles while you just need to do it for a specific one, at least that's what I think based on the ticket. You do join with `test_result_profiles` but not limit the scope against any `profile`. Also you don't select the `latest` test results for the given profile either :thinking:

Reply:
Also the tests are wrong too, because you want to sort based on ALL the test results not just the profile, you need to also filter the GQL results using `with_results_for_policy_id`.